src/wkt/src/timestamp.rs

```diff
@@ -447,6 +487,41 @@ impl TryFrom<Timestamp> for chrono::DateTime<chrono::Utc> {
     }
 }
 
+/// Converts from [Timestamp] to [std::time::SystemTime].
+///
+/// This conversion may fail if the [std::time::SystemTime] value is out of
+/// range.
+///
+/// # Example
+/// ```
+/// # use std::time::SystemTime;
+/// # use google_cloud_wkt::{Timestamp, TimestampError};
+/// let ts = Timestamp::new(0, 0)?;
+/// let epoch = SystemTime::try_from(ts)?;
+/// assert_eq!(epoch, SystemTime::UNIX_EPOCH);
+/// # Ok::<(), anyhow::Error>(())
+/// ```
+impl TryFrom<Timestamp> for std::time::SystemTime {
+    type Error = TimestampError;
+    fn try_from(value: Timestamp) -> Result<std::time::SystemTime, Self::Error> {
+        let mut ts = std::time::SystemTime::UNIX_EPOCH;
+        let s = value.seconds();
+        if s >= 0 {
+            let d = std::time::Duration::from_secs(s as u64);
+            ts = ts
+                .checked_add(d)
+                .ok_or_else(|| TimestampError::OutOfRange)?;
+        } else {
+            let d = std::time::Duration::from_secs(-s as u64);
+            ts = ts
+                .checked_sub(d)
+                .ok_or_else(|| TimestampError::OutOfRange)?;
+        }
```

Comment:
```suggestion
        let s = value.seconds();
        let ts = if s >= 0 {
            let d = std::time::Duration::from_secs(s as u64);
            [ts = ts](std::time::SystemTime::UNIX_EPOCH)
                .checked_add(d)
                .ok_or_else(|| TimestampError::OutOfRange)?
        } else {
            let d = std::time::Duration::from_secs(-s as u64);
            std::time::SystemTime::UNIX_EPOCH
                .checked_sub(d)
                .ok_or_else(|| TimestampError::OutOfRange)?
        };
```

I am not sure it is better...